cfgov/unprocessed/apps/regulations3k/js/permalinks-utils.js

```diff
@@ -74,12 +82,125 @@ const getCurrentParagraph = ( currentPosition, paragraphs ) => {
  * @returns {object} window object
  */
 const updateUrlHash = () => {
-  const currentParagraph = getCurrentParagraph( scrollY(), paragraphPositions );
+  const currentParagraph = getCurrentParagraph( scrollY() + wayfinderOffset, paragraphPositions );
   // Setting the window state to `.` removes the URL hash
   const hash = currentParagraph ? `#${ currentParagraph }` : '.';
   return window.history.replaceState( null, null, hash );
 };
 
+/**
+ * getCommentMarker - Does the legwork for the more complex comment markers
+ *
+ * @param {string} label - data-label of the current paragraph
+ *
+ * @returns {string} formatted comment marker
+ */
+const getCommentMarker = label => {
+  let commentedSection;
+  let commentedParagraph;
+  let commentParagraph;
+  const splitCurrentParagraph = label.split( 'Interp' );
+  const commentedParagraphID = splitCurrentParagraph[0].split( '-' );
+  const commentParagraphID = splitCurrentParagraph[1]
+    .split( '-' );
+  commentedSection = commentedParagraphID[0];
+  if ( commentedSection.match( /[A-Z]/ ) ) {
+    commentedSection = 'app. ' + commentedParagraphID[0];
+    commentedParagraph = '';
+  } else {
+    commentedParagraph = commentedParagraphID
+      .slice( 1, -1 )
+      .join( ')(' );
+    commentedParagraph = '(' + commentedParagraph + ')';
+  }
+  commentParagraph = commentParagraphID
+    .slice( 1 )
+    .join( '.' );
+  if ( commentParagraph !== '' ) {
+    commentParagraph = '-' + commentParagraph;
+  }
+
+  return commentedSection + commentedParagraph + commentParagraph;
+};
+
+/**
+ * getWayfinderInfo - process paragraph to create wayfinder
+ * @param {string} label - label of current paragraph
+ * @param {string} sectionTitle - title of current section
+ *
+ * @returns {object} object of the values for wayfinder
+ */
+const getWayfinderInfo = ( label, sectionTitle ) => {
+  let sectionFormattedTitle;
+  let paragraphMarker;
+
+  // For interpretations, the wayfinder should look like "Comment 4(a)-1.iv.A"
+  // Or like "Comment app. G-1.iv.A" for interpretations of appendices
+  if ( sectionTitle.indexOf( 'Comment for ' ) > -1 || label.indexOf('Interp') > -1 ) {
+    sectionFormattedTitle = 'Comment ';
+    paragraphMarker = getCommentMarker( label );
+  } else if ( sectionTitle.indexOf( 'Appendix ' ) === 0 ) {
+    // For appendices, the wayfinder should look like "Appendix A"
+    sectionFormattedTitle = sectionTitle.match( wayfinderRegex.appendixTitle )[0];
```

Comment:
So it turns out we have the occasional interpretation section that applies to *two* appendices at once, like `/policy-compliance/rulemaking/regulations/1026/2020-04-01/Interp-GH/`. That produces `TypeError: null is not an object (evaluating 't.match(o)[0]')` here.

Reply:
Turns out this particular case was being caused by some unexpected, incorrect paragraph markers. But the updates to handle errors are working, and all the oddball, multi-letter interps and appendices I can find have wayfinders that are working as well as the underlying data allows now.